src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/IDocumentFilter.cs

```diff
@@ -9,6 +10,11 @@ public interface IDocumentFilter
         void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context);
     }
 
+    public interface IAsyncDocumentFilter : IDocumentFilter
+    {
+        Task ApplyAsync(OpenApiDocument swaggerDoc, DocumentFilterContext context);
+    }
+
```

Comment:
Similar to my point above, I'm wondering about the name for the new interfaces. Prefixing with `Async` feels a little unconventional because really the standard is to postfix the async methods as opposed to the interface itself. Of course it can't be the same as the original name but perhaps we could move to something like `IOpenApiDocumentFilter`, `IOpenApiOperationFilter` etc.?

Reply:
Yes it's a bit strange, but it's what MSFT does: for example see [IAsyncEnumerator](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.iasyncenumerator-1?view=dotnet-plat-ext-3.1) with the MoveNextAsync Method.
But of course I can call it something like IOpenApiDocumentFilter (I like this one)

Reply:
Maybe we could mark IDocumentFilter with [Obsolete] then